File: Reverse Linked List.c
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* post = head;
    struct ListNode* pre = NULL;
    struct ListNode* cur = NULL;
    
    while(post != NULL)
    {
        pre = cur;
        cur = post;
        post = post -> next;
        cur -> next =pre;
    }
    
    return cur;
}

File: Reverse Linked List.js
var reverseList = function(head) {
    var post = head;
    var pre = null;
    var cur = null;
    
    while(post != null)
    {
        pre = cur;
        cur = post;
        post = post.next;
        cur.next = pre; 
    }
    return cur;
};
